Modules/Core/SpatialObjects/include/itkImageMaskSpatialObject.hxx

```diff
@@ -39,6 +43,12 @@ ImageMaskSpatialObject<TDimension, TPixel>::IsInsideInObjectSpace(const PointTyp
 
   const IndexType index = image->TransformPhysicalPointToIndex(point);
 
+  if (this->m_UseMaskValue == true)
+  {
+    return Superclass::GetBufferedRegion().IsInside(index) &&
+           Math::ExactlyEquals(image->GetPixel(index), this->m_MaskValue);
+  }
+
   return Superclass::GetBufferedRegion().IsInside(index) && Math::NotExactlyEquals(image->GetPixel(index), PixelType{});
```

Comment:
Avoid unnecessary computations when outside the image, or m_useMaskValue == false.  Utilize early termination on false conditions in compound conditional.


```c++
constexpr auto background_indicator_zero = PixelType{};

const bool is_inside = Superclass::GetBufferedRegion().IsInside(index);
return  is_inside && 
  (
  (this->m_useMaskValue == false) &&  Math::NotExactlyEquals(image->GetPixel(index),background_indicator_zero)
) && 
Math::ExactlyEquals(image->GetPixel(index), this->m_MaskValue)
;
```

Reply:
Thanks!   Had to tweak the logic a bit...

  return is_inside && ((m_UseMaskValue == true && 
      Math::ExactlyEquals(image->GetPixel(index), this->m_MaskValue))
      || (m_UseMaskValue == false &&
      Math::NotExactlyEquals(image->GetPixel(index), background_zero)));

If you can think of a faster way, let me know.